src/Symfony/Component/Asset/VersionStrategy/CachedVersionStrategy.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\VersionStrategy;
+
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+/**
+ * Cache generated versions and path of any version strategy.
+ */
+class CachedVersionStrategy implements VersionStrategyInterface
+{
+    private $strategy;
+
+    private $cache;
+
+    private $options;
+
+    /**
+     * @param array $options
+     *
+     * List of available options:
+     *  * prefix: Begin the cache key
+     *  * beta: A float that, as it grows, controls the likeliness of triggering
+     *   early expiration. 0 disables it, INF forces immediate expiration.
+     *  * expiresAfter: The period of time from the present after which the item MUST
+     *   be considered expired. An integer parameter is understood to be the time in
+     *   seconds until expiration.
+     *
+     * @see CacheInterface::get()
+     * @see ItemInterface::expiresAfter()
+     */
+    public function __construct(VersionStrategyInterface $strategy, CacheInterface $cache, array $options = [])
+    {
+        $this->strategy = $strategy;
+        $this->cache = $cache;
+        $this->options = array_replace([
+            'prefix' => 'asset_',
+            'beta' => null,
+            'expiresAfter' => null,
+        ], $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getVersion(string $path)
+    {
+        return $this->cache->get($this->getCacheKey('version_'.$path), function (ItemInterface $item) use ($path) {
+            $item->expiresAfter($this->options['expiresAfter']);
+
+            return $this->strategy->getVersion($path);
+        }, $this->options['beta']);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function applyVersion(string $path)
+    {
+        return $this->cache->get($this->getCacheKey('path_'.$path), function (ItemInterface $item) use ($path) {
```

Comment:
`return $this->cache->get('p_'.rawurlencode($path), [$this->strategy, 'applyVersion']);`